rewrite web component front end into TV-Show style / Piscine style

SallyMcGrath opened the pull request:
- rm web components and event driven self contained components
- rewrite into style of TV show, "vanilla React"
- entirely state driven 
- router and views for Ali
- localStorage cache

I took out the tests for now

- the back end needs a search api for hashtags  - 

Daniel do you like this structure better? if so, will you build out the rest of the back end and add in anything else you want and I will finish and polish according to your desires

illicitonion commented:
Yeah this shape is looking a lot better, thank you!

I've been doing a little poking, and it seems like a lot of response -> state things aren't super smooth right now? e.g. signing up as an existing user, or logging in with the wrong password get you stuck in a broken state? And after following someone I needed to log out and log back in to actually see that take effect?

But generally looking great. Is there anything you particularly need backend-wise to unblock you? I think functionally the only thing missing is a "Be able to click a hashtag to see matching blooms" endpoint? Otherwise I think it's all just clean-up/persistence/tests?

(And then introducing all of the intentional bugs)

SallyMcGrath commented:
> Yeah this shape is looking a lot better, thank you!
> 
> I've been doing a little poking, and it seems like a lot of response -> state things aren't super smooth right now? e.g. signing up as an existing user, or logging in with the wrong password get you stuck in a broken state? And after following someone I needed to log out and log back in to actually see that take effect?
> 
> But generally looking great. Is there anything you particularly need backend-wise to unblock you? I think functionally the only thing missing is a "Be able to click a hashtag to see matching blooms" endpoint? Otherwise I think it's all just clean-up/persistence/tests?
> 
> (And then introducing all of the intentional bugs)

yeah I've been tinkering around with the errors, which I think will need to be super clear, that's why it's a bit janky. But they aren't big issues, just need running through and tightening up and triggering redraws where suitable, which I will do now if you're happy with the design. 

I will fix and polish all that and, yes, I would like from you the following:

- hashtag endpoint, ideally with hashtags already within as html
- some notes on patterns they can look for in the back end
- a db if you can be bothered - if not, a richer set of seed data so it's clearer to them 

The prep has 9 short simple sections, like this

1. What is legacy code
2. Fear and logic: Chestertons' Fence, your code is a crime scene
3. Patterns: what are design patterns, name some patterns, what is chunking
4. Finding things: navigating files, tracing a request
5. Errors and signals : trace an error through the stack
6. Debugging: use the debugger and logic to chase down a bug
7. Testing: write a playwright.spec using GWT
8. Fixing : activity to fix one bug (more in coursework)
9. Extending : activity to add an unfollow button (more features in coursework)

I have written (and rewritten as I rewrote the codebase and had that useful discussion) the bumf around this but I am still deciding what specific error to cause, trace, debug, test, and fix. I think I definitely want 1 example to take through each section, not lots of different examples.

SallyMcGrath commented:
Oh and my next question was, because I don't really have it clear in my head yet - what do you want to bring from their decomposition project into this?
